src/databases/RAW/avtRAWWriter.C

```diff
@@ -197,10 +203,19 @@ avtRAWWriter::WriteChunk(vtkDataSet *ds, int chunk)
     if(nDomains > 1)
         fprintf(file, "Object%d\n", chunk + 1);
 
+    vtkIdType nids;
```

Comment:
I would almost make the same observation here as well. One begins with `while` and the other with `for`. But, it is protecting a bigger and more complex loop body too. So, I am fine with this one as is.

Reply:
I should say, I am fine with the other cases too. I just wanted to make the suggestion that sometimes the conditional compilation logic to avoid wholly duplicating a small loop body can be more trouble than its worth. That goes away as the loop body gets bigger and/more complex.